Adding third party module - example

augustocollado opened the pull request:
### Description

This PR completes the third-party module article by illustrating the process of adding a third-party pallet, the error you'll get when trying to compile, and how to apply the solution by patching the deps with dinner.
This section relies on this other repo containing the pallet: https://github.com/papermoonio/pallet-toggle

### Checklist

- [x] I have added a label to this PR 🏷️
- [x] If this page requires a disclaimer, I have added one
- [x] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects

albertov19 commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+#### Make the Standard Features Available to the Compiler {: #add-standard-features }
+
+Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which is located in the folder `container-chains/templates/simple/runtime`.
```
```suggestion
Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which for this example that uses the Tanssi GitHub repo, is located in the folder `container-chains/templates/simple/runtime`.
```

albertov19 commented on builders/build/customize/adding-external-module.md:
````diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+]
+```
+
+#### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
````
```suggestion
### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
```

albertov19 commented on builders/build/customize/adding-external-module.md:
````diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+);
+```
+
+### Compile {: #compile }
````
```suggestion
### Compile Runtime {: #compile-runtime }
```

albertov19 commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+#### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
+
+Add the following snippet that configures the module and add the module within the `construct_runtime!`macro.
```
```suggestion
Next, add the following snippet in the `lib.rs` file inside the runtime folder. This configures the module and add the module within the `construct_runtime!`macro.
```

albertov19 commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,121 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+### Patch depedencies {: #patch-dependencies }
+
+Finally, executing the `diener` [command](#solving-dependencies-conflicts-diener) will add a `patch` section to your workspace `Cargo.toml` file, overriding the dependencies and unifying origins and versions.
```
Add an example of how the diener command is executed for this example

albertov19 reviewed (changes requested):
Left some feedback to improve readability :)

albertov19 reviewed (changes requested):
Looks great! As my last piece of feedback, I would use now `termynal` 

It will be an easy thing to adapt here. Check the reference PR -> https://github.com/moondance-labs/tanssi-docs/pull/74

eshaben commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+## Example of the Double Reference Issue {: #double-reference-issue }
+
+To illustrate the situation, the following steps add a demo [external module](https://github.com/papermoonio/pallet-toggle.git){target=_blank} to a custom runtime based on the [baseline Appchain template](/builders/build/templates/substrate/){target=_blank}. One way to follow this tutorial is to clone the [Tanssi Github repository](https://github.com/moondance-labs/tanssi){target=_blank}, which w ...
```
All of the `target=_blank`s on this page should be changed to `target=\_blank`

eshaben commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+The first step to reproduce the double reference issue is to declare the dependency in the `Cargo.toml` file located in the repository's root folder, under the section `[dependencies]`. For this example, a simple [toggle module](https://github.com/papermoonio/pallet-toggle.git){target=_blank} is used. 
+
+This `toggle` module, built for testing and educational purposes, adds a basic logic to the runtime, allowing users to switch a state between true and false.
```
```suggestion
This `toggle` module, built for testing and educational purposes, adds basic logic to the runtime, allowing users to switch a state between true and false.
```

eshaben commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+### Make the Standard Features Available to the Compiler {: #add-standard-features }
+
+Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which for this example that uses the Tanssi GitHub repo, is located in the folder `container-chains/templates/simple/runtime`.
```
```suggestion
Having declared the module in the workspace `Cargo.toml` file, the dependency can now be added to the specific template `Cargo.toml` file, which, for this example that uses the Tanssi GitHub repo, is located in the folder `container-chains/templates/simple/runtime`.
```

eshaben commented on builders/build/customize/adding-external-module.md:
```diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+
+### Configure and Add the Module to the Runtime {: #configure-module-in-the-runtime }
+
+Next, add the following snippet in the `lib.rs` file inside the runtime folder. This configures the module and add the module within the `construct_runtime!`macro.
```
```suggestion
Next, add the following snippet to the `lib.rs` file inside the runtime folder. This configures the module and adds the module within the `construct_runtime!` macro.
```

eshaben commented on builders/build/customize/adding-external-module.md:
````diff
@@ -29,3 +29,135 @@ If the `diener` executable file, the cloned [Polkadot SDK repository](https://gi
+error: failed to select a version for `syn`.
+```
+
+### Patch depedencies {: #patch-dependencies }
````
```suggestion
### Patch Dependencies {: #patch-dependencies }
```

albertov19 commented:
@augustocollado you can remove the terminal image then :)

augustocollado commented:
> @augustocollado you can remove the terminal image then :)

Ouch! I deleted the image, but I don't know how that copy with that name got in that new folder I created. Fixed.

> @augustocollado don't forget the `target=\_blank` as per the changes made in this PR: #76

Sorry about that, I thought I had gotten them all :S

eshaben reviewed (approved):
LGTM!

albertov19 reviewed (approved):
LGTM
